Challenge3/src/utils.ts

```diff
@@ -0,0 +1,87 @@
+import { Finding, FindingSeverity, FindingType, ethers, Alert, AlertsResponse, getAlerts } from "forta-agent";
+import { L2_DAI_TOKEN_ADDRESS, L2_TOKEN_ABI } from "./constants";
+import { Provider } from "@ethersproject/providers";
+
+export const l1Finding = (_l1OptEscrowBalance: string, _l1ArbEscrowBalance: string): Finding => {
+  return Finding.fromObject({
+    name: `Maker L2 Bridge: DAI Locked in L1 Escrow Contracts for Optimism and Arbitrum`,
+    description: `DAI locked for Optimism: ${_l1OptEscrowBalance}. DAI locked for Arbitrum: ${_l1ArbEscrowBalance}.`,
+    alertId: "L1-DAI-LOCKED",
+    severity: FindingSeverity.Info,
+    type: FindingType.Info,
+    protocol: "Ethereum",
+    metadata: {
+      l1OptEscrowBalance: _l1OptEscrowBalance,
+      l1ArbEscrowBalance: _l1ArbEscrowBalance,
+    },
+  });
+};
+
+export const l2Finding = (_l1DaiLocked: string, _l2DaiSupply: string, _chainId: number): Finding => {
+  const chainName = _chainId === 10 ? "Optimism" : "Arbitrum";
+  return Finding.fromObject({
+    name: `Maker L2 Bridge: DAI Locked in L1 Escrow Contract for ${chainName} is greater than L2 DAI Supply`,
+    description: `L1 DAI Locked: ${_l1DaiLocked}. L2 (${chainName}) DAI Supply: ${_l2DaiSupply}.`,
+    alertId: "BRIDGE-BREACHED",
+    severity: FindingSeverity.High,
+    type: FindingType.Exploit,
+    protocol: chainName,
+    metadata: {
+      l1DaiLocked: _l1DaiLocked,
+      l2DaiSupply: _l2DaiSupply,
+    },
+  });
+};
+
+export const emitL1Alert = (_l1OptEscrowBalance: string, _l1ArbEscrowBalance: string) => {
+  const l1DaiLockedChangeAlert: Alert = {
+    alertId: "L1-DAI-LOCKED-CHANGE",
+    hasAddress: () => false,
+    metadata: {
+      l1OptEscrowBalance: ethers.BigNumber,
+      l1ArbEscrowBalance: ethers.BigNumber,
+    },
+  };
+  const l1DaiLockedChangeAlertsResponse: AlertsResponse = {
+    alerts: [l1DaiLockedChangeAlert],
+    pageInfo: { hasNextPage: false },
+  };
+  l1DaiLockedChangeAlertsResponse.alerts[0].metadata.latestL1OptEscrowBalance = _l1OptEscrowBalance;
+  l1DaiLockedChangeAlertsResponse.alerts[0].metadata.latestL1ArbEscrowBalance = _l1ArbEscrowBalance;
+};
+
+const getL1Alerts = async (chainId: number): Promise<AlertsResponse> => {
+  return await getAlerts({
+    alertId: "L1-DAI-LOCKED-CHANGE",
+    chainId: chainId,
+  });
+};
+const getMockL1Alerts = async (chainId: number): Promise<AlertsResponse> => {
+  return {
+    alerts: [
+      {
+        alertId: "L1-DAI-LOCKED-CHANGE",
+        hasAddress: () => false,
+        metadata: {
+          l1OptEscrowBalance: "1000",
+          l1ArbEscrowBalance: "2000",
+        },
+      },
+    ],
+    pageInfo: { hasNextPage: false },
+  };
+};
+
+export const getL1DaiLocked = async (l2ChainId: number): Promise<ethers.BigNumber> => {
+  const l1Alerts = await getMockL1Alerts(l2ChainId);
```

Comment:
At the moment you're mocking the `getAlerts` function inside the production code and not in the unit tests, which is wrong.

The way to mock it properly (while using the 'real' one in prod) is to pass the function as a parameter, starting from the `provideHandleBlock`

In the prod code, you should then pass the real function, and in the tests the mock one